src/ui/src/components/core/content/CoreDataframe/CoreDataframeCellText.vue

```diff
@@ -21,6 +21,38 @@ function stopEditing() {
 	if (newValue === props.value) return;
 	emits("change", textarea.value.value);
 }
+
+function autoResizeFromRef() {
+	if (!textarea.value?.ref) return;
+
+	const el = textarea.value.ref as HTMLTextAreaElement;
+
+	const style = getComputedStyle(el);
+	if (!style) return;
+
+	el.style.height = "auto";
+
+	const borderTop = parseFloat(style.borderTopWidth || "0");
+	const borderBottom = parseFloat(style.borderBottomWidth || "0");
+	const verticalBorder = borderTop + borderBottom;
+
+	el.style.height = `${el.scrollHeight + verticalBorder}px`;
+}
+
+onMounted(() => {
+	if (props.wrapText && props.editable) {
+		nextTick(() => autoResizeFromRef());
+	}
+});
+
+watch(
+	() => props.editable || props.value,
+	() => {
+		if (props.wrapText && props.editable) {
+			nextTick(() => autoResizeFromRef());
+		}
+	},
+);
```

Comment:
1. you can use `immediate: true` to execute when mounted
2. you use the callback props to refactor a bit
```suggestion
watch(
	() => props.editable && props.value,
	(shouldResize) => {
		if (shouldResize) {
			nextTick(() => autoResizeFromRef());
		}
	},
	{ immediate: true }
);
```